src/core/composer/qgscomposermap.cpp

```diff
@@ -153,6 +153,7 @@ QgsMapSettings QgsComposerMap::mapSettings( const QgsRectangle &extent, QSizeF s
   jobMapSettings.setOutputDpi( dpi );
   jobMapSettings.setBackgroundColor( Qt::transparent );
   jobMapSettings.setRotation( mEvaluatedMapRotation );
+  jobMapSettings.setEllipsoid( renderCrs.ellipsoidAcronym() );
```

Comment:
Should be

jobMapSettings.setEllipsoid( mComposition->project()->ellipsoid() );

(Remember - ellipsoid choice is NOT locked to CRS choice. QGIS suggests that it should be by auto setting the ellipsoid in the project properties dialog when the project CRS changes, but it's not enforced and users may override this if they want)